### PR TITLE
Add forum settings

### PR DESCRIPTION
**Pull request by xzhflying**

### CODE REVIEW

**Comment by allenwq** [comment]
No sure if we want to customize topic title as well...

@lowjoel  What do you think ?  Other places look good.

File: app/models/course/forum_settings.rb [hunk 1]
```diff
@@ -27,18 +28,47 @@ def title=(title)
     @settings.title = title
   end
 
+  # Returns the title of topics in forums component
+  #
+  # @return [String] The custom or default title of topics in forums component
+  def topic_title
```
Comment (lowjoel) [inline app/models/course/forum_settings.rb]: what's this for?

File: app/models/course/forum_settings.rb [hunk 2]
```diff
@@ -27,18 +28,47 @@ def title=(title)
...
+    title = nil unless title.present?
+    @settings.topic_title = title
+  end
+
   # Returns the forum pagination count
   #
   # @return [Integer] The pagination count of forum
-  def pagination
-    @settings.pagination || 50
+  def forum_pagination
```
Comment (lowjoel) [inline app/models/course/forum_settings.rb]: Why change the name?

File: app/models/course/forum_settings.rb [hunk 3]
```diff
@@ -27,18 +28,47 @@ def title=(title)
...
-  def pagination=(count)
-    @settings.pagination = count
+  def forum_pagination=(count)
+    @settings.forum_pagination = count
+  end
+
+  # Returns the topic pagination count
+  #
+  # @return [Integer] The pagination count of topic
+  def topic_pagination
```
Comment (lowjoel) [inline app/models/course/forum_settings.rb]: I don't think you can meaningfully paginate a threaded conversation.

File: app/models/course/forum_settings.rb (hunk 1 above)
Comment (lowjoel) [inline app/models/course/forum_settings.rb]: This is for changing what the forum calls a topic? I don't think that's necessary.

File: app/models/course/forum_settings.rb (hunk 3 above)
Comment (xzhflying) [inline app/models/course/forum_settings.rb]: So... I just need one pagination and use it in both forum and forum::topic?

Comment (lowjoel) [inline app/models/course/forum_settings.rb]: You cannot paginate a threaded conversation.

Comment (xzhflying) [inline app/models/course/forum_settings.rb]: OK...I'll remove topic related settings.

Comment (allenwq) [inline app/models/course/forum_settings.rb]: I think topic_pagination is for topic index page, and forum_pagination is for forum index page ?

Comment (xzhflying) [inline app/models/course/forum_settings.rb]: Yes, so is this necessary?

Comment (lowjoel) [inline app/models/course/forum_settings.rb]: I don't think the forum index page is paginated. How many forums can a person possibly create?

Comment (allenwq) [inline app/models/course/forum_settings.rb]: OK, not that many actually, so let's remove this. And then the forum pagination is for Topic.

BTW, for post pagination, my concern is that if there are too many posts in one thread, the page will become very slow to load or even unable to load (v1 once got the similar issue).  We need do some thing about this ?

Comment (lowjoel) [inline app/models/course/forum_settings.rb]: If you know how to logically paginate a threaded conversation, I'm all ears.

**Pull request closed without merging**